src/Components/Web.JS/src/Services/ComponentDescriptorDiscovery.ts

```diff
@@ -26,6 +26,35 @@ function discoverServerComponents(document: Document): ServerComponentDescriptor
   return discoveredComponents.sort((a, b): number => a.sequence - b.sequence);
 }
 
+const blazorStateCommentRegularExpression = /\W*Blazor-Component-State:(?<state>[a-zA-Z0-9\+\/=]+)$/;
+
+export function discoverPersistedState(node: Node): string {
```

Comment:
Could this return `string | null`? I'm not sure why we're coalescing with `''` below - it's odd to use empty string as the absence of a value.

Reply:
I've changed it, the reason was that it got passed to mono_js_string_to_string and that requires a string and it was less hassle to just return a value and avoid null/undefined all-together